src/Tokenizer.js

```diff
@@ -101,15 +103,16 @@ export function token (type) {
 
 /**
  * @param {string} value
- * @return {any[]}
+ * @return {[]}
```

Comment:
won't this be an equivalent to `never[]` or something? from what I remember this allocates `Element[]`, I think this wouldn't compile correctly when declared like this when using TS, not sure what about JSDoc though

Reply:
Hmm. It returns _exactly_ `[]`. Not sure what it's going to be used for afterwards but it's `[]` at this point

Reply:
Ok fair. I'll use `unknown[]`. Good call

Reply:
In a similar fashion `unknown[]` is not a good fit here because it would require either casting or type refining down the road - we want the result of `alloc` to be consumable by other code. That's why generic (`T[]`) would be preferred here.

Look what happens with `unknown`:
https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAQwDarhAFASgFyLgDWYcA7mANoC6iA3gFCLOIBOAplCK0jQwL4MGUAJ4AHdogBKIVJIC89URLwByVrPaqANIgBuaEOwIBnKKxhgA5oIAm7CKmQdEEBGbYEZcoQdYp0THZbREU0DGwcBnCg2wA6MRATAAssViiGNzAPVgAmL01QgIjgygAGaiEgA